pkg/coscheduling/coscheduling.go

```diff
@@ -55,15 +55,31 @@ var _ framework.PostBindPlugin = &Coscheduling{}
 
 const (
 	// Name is the name of the plugin used in Registry and configurations.
-	Name = "Coscheduling"
+	Name                                       = "Coscheduling"
+	defaultPermitWaitingTimeSeconds      int64 = 60
+	defaultDeniedPGExpirationTimeSeconds int64 = 20
 )
 
-// New initializes a new plugin and returns it.
+// New initializes and returns a new Coscheduling plugin.
 func New(obj runtime.Object, handle framework.FrameworkHandle) (framework.Plugin, error) {
-	args, ok := obj.(*config.CoschedulingArgs)
-	if !ok {
-		return nil, fmt.Errorf("want args to be of type CoschedulingArgs, got %T", obj)
+	var (
+		args *config.CoschedulingArgs
+		ok   bool
+	)
+	if obj != nil {
+		args, ok = obj.(*config.CoschedulingArgs)
+		if !ok {
+			return nil, fmt.Errorf("want args to be of type CoschedulingArgs, got %T", obj)
+		}
+	}
+	// TODO: Replace it with v1beta1.SetDefaultsCoschedulingArgs
+	if args.PermitWaitingTimeSeconds == 0 {
+		args.PermitWaitingTimeSeconds = defaultPermitWaitingTimeSeconds
+	}
+	if args.DeniedPGExpirationTimeSeconds == 0 {
+		args.DeniedPGExpirationTimeSeconds = defaultDeniedPGExpirationTimeSeconds
```

Comment:
Same as above, no need to add duplicated defaulting logic.

Reply:
> Same as above, no need to add duplicated defaulting logic.

Similarly, if we use the version`v1beta1,` we won't need them.

Reply:
Again, I'm not following what do you mean by "if we switch...".